packages/experimental/Badge/src/Badge.tsx

```diff
@@ -39,25 +41,28 @@ export const Badge = compose<BadgeType>({
     const Slots = useSlots(userProps, (layer) => badgeLookup(layer, userProps));
 
     return (final: BadgeProps, ...children: ReactNode[]) => {
-      const { icon, iconPosition, size, ...mergedProps } = mergeProps(badge, final);
+      const { icon, iconPosition, size, shadow, ...mergedProps } = mergeProps(badge, final);
       const showContent = size !== 'tiny' && size !== 'extraSmall';
       const showIcon = size !== 'tiny';
+      const BadgeComponent = shadow ? Slots.shadow : Fragment;
```

Comment:
If I understand correctly, in this case client will need to pass shadowToken, right?

Reply:
It will be set by slotprops. If slotprops doesn't set a shadowToken, then <Slots.shadow> will only render it's children. You can look at shadow.tsx to see the implementation